users/api/auth.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package api
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/absmach/magistrala"
+	"github.com/absmach/magistrala/internal/api"
+	"github.com/absmach/magistrala/pkg/apiutil"
+	"github.com/absmach/magistrala/pkg/auth"
+	"github.com/absmach/magistrala/pkg/errors"
+	svcerr "github.com/absmach/magistrala/pkg/errors/service"
+	"github.com/absmach/magistrala/pkg/policies"
+	"github.com/go-kit/kit/endpoint"
+)
+
+const sessionKey = "session"
+
+type authEndpointFunc func(interface{}) (*magistrala.AuthorizeReq, error)
+
+func identifyMiddleware(authClient auth.AuthClient) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			token := apiutil.ExtractBearerToken(r)
+			if token == "" {
+				api.EncodeError(r.Context(), apiutil.ErrBearerToken, w)
+				return
+			}
+
+			resp, err := authClient.Identify(r.Context(), &magistrala.IdentityReq{Token: token})
+			if err != nil {
+				api.EncodeError(r.Context(), err, w)
+				return
+			}
+
+			ctx := context.WithValue(r.Context(), sessionKey, auth.Session{
```

Comment:
Please fix the linter warning. Since context is a generic construct in Go, it's strongly advised to create your own type for the key:
```go
type sessionKeyType string

const sessionKey = sessionKeyType("session")

```